### ISSUE TITLE
getAttribute returns undefined

### ISSUE THREAD

**Issue by johnsickels** [flocasts/flagpole#59]
`getAttribute` to return some value, but is returning `undefined` instead

Repro:
```
suite
  .scenario("Homepage Loads", "browser")
  .open("https://flocasts.github.io/flagpole")
  .next(async (context) => {
    const foo = await context.exists(".content h1 a");
    context.assert(await foo.hasAttribute("data-id")).equals(true);
    context
      .assert(await foo.getAttribute("data-id"))
      .equals("what-is-flagpole");
  });
```

Flagpole version: 2.5.18

**Comment by jasonbyrne** [comment]
Looking into this one, but seems to be a bit more complex will take me a bit to figure out where it breaks down. Seems to have something to do with it being a data attribute, I think, since href works

**Comment by johnsickels** [comment]
I had some thoughts about this one, because I thought it had something to do with the data attr as well... If I do this:

```
const foo = await context.find(".content h1 a");
console.log(foo);
context.comment(await foo.getAttribute("data-id"));
```

the console.log with show:
```
BrowserElement {
  _sourceCode: '<a href="#/?id=what-is-flagpole" data-id="what-is-flagpole" class="anchor"><span>What is Flagpole?</span></a>',
  _input: ElementHandle {
```

but the `.getAttribute("data-id")` still returns `undefined`. 

Naturally, I then looked for something like `.getData(key)` since `.hasData(key)` is already present.

**Comment by jasonbyrne** [comment]
Yes, I saw the undefined thing as well. It is casting undefined to a string, so that's why. So I'll fix that, but I need to also figure out how to get those data attributes.

**Comment by jasonbyrne** [comment]
Got this fixed in v2.5.19
